server/routerlicious/packages/lambdas-driver/src/document-router/documentPartition.ts

```diff
@@ -18,7 +18,7 @@ import { DocumentContext } from "./documentContext";
 
 export class DocumentPartition {
 	private readonly q: QueueObject<IQueuedMessage>;
-	private readonly lambdaP: Promise<IPartitionLambda>;
+	private readonly lambdaP: Promise<IPartitionLambda> | Promise<void>;
```

Comment:
Do we still need to keep `Promise<IPartitionLambda>`?

Reply:
Yes, in line 131 `this.lambdaP.then((lambda) => { lambda.close(closeType);})` because when we call `close()` we check if `lambdaP` is not void and call `close()` to safely close it, so there are cases where this is not void if the `.then()` did not complete